robot-code/micro-python/client_udp_listen.py

```diff
@@ -1,11 +1,78 @@
 import socket
+import network
+import time
 
+ROBOT_NAME = "St. Francis"
+SSID = "WATCHTOWER"
+PASSWORD = "lancerrobotics"
 PORT = 2367
-s=socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-s.bind(('', PORT)) #binding the port
-print('waiting...')
+my_game_state = 0
 
-while True:   #receiving data
-  data,addr=s.recvfrom(1024)
 
-print('received:',data,'from',addr)
\ No newline at end of file
+### Called by user to receive the data from UDP
+### The output either will show the updated game state or the controller status depending on game state
+def read_udp_packet(data):
+	data = None
+	while data == None:
+		data, addr = s.recvfrom(1024)
+
+	if len(data) == 24:
+		return parse_robot_command(data)
+
+	elif len(data) == 65:
+		return parse_game_state(data)
+
+
+### Parses the UDP data to apply the game state to the robot if applicable
+### Returns a message with the game state for user info, no action required with this information
+def parse_game_state(data):
+	hex_list = [hex(byte) for byte in data]
+	game_state = data.pop(0)
+
+	#find out if the game state command is applicable to this robot
+	for elem in [hex_list[i:i+16] for i in range(0, len(hex_list), 16)]:
```

Comment:
double for loop nesting in one line is bad for readability and makes this comment hard. You included the first byte of the message in the range so all the names will be off by one.